// Protocol Buffers for Saturnin Firebird-log Service
// Copyright 2019 Firebird Project.  All rights reserved.
// https://www.firebirdsql.org/
// https://saturnin.rtfd.io
//
// License: The MIT License
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"),
// to deal in the Software without restriction, including without limitation
// the rights to use, copy, modify, merge, publish, distribute, sublicense,
// and/or sell copies of the Software, and to permit persons to whom
// the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included
// in all copies or substantial portions of the Software.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

syntax = "proto3";

package saturnin.fblog;

import public "google/protobuf/struct.proto";
import public "google/protobuf/timestamp.proto";
import public "saturnin/protobuf/saturnin.proto";

// Firebird Log Entry

message FirebirdLogEntry {
  string source_id = 1 ;                    //
  google.protobuf.Timestamp timestamp = 2 ; //
  string message = 3 ;                      //
}

// Log source

message LogSource {
  oneof source {
    string filespec = 1 ;
    google.protobuf.Struct server = 2 ;
    saturnin.DataPipe pipe = 3 ;
  }
}

// MONITOR Message Data

message RequestMonitor {
  LogSource source = 1 ;           // [required] Source for log entries
  saturnin.DataPipe push_to = 2 ;  // Optional data pipe specification
}

message ReplyMonitor {
  int32 handle = 1 ;           // Handle assigned to monitoring request
  saturnin.DataPipe pipe = 2 ; // Optional data pipe specification
}

// STOP_MONITOR Message Data

message RequestStopMonitor {
  int32 handle = 1 ; // [required] Handle assigned to monitoring request
}

// ENTRIES Message Data

message RequestEntries {
  LogSource source = 1 ;           // [required] Source for log entries
  saturnin.DataPipe push_to = 2 ;  // Optional data pipe specification
}


